.circleci/osx_install_dependencies.sh

```diff
@@ -59,27 +59,47 @@ then
   brew install wget
   brew install coreutils
   brew install diffutils
-  ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+  # writing to /usr/local/lib need administrative privileges.
+  sudo ./scripts/install_obsolete_jsoncpp_1_7_4.sh
 
   # z3
   z3_version="4.12.1"
   z3_dir="z3-${z3_version}-x64-osx-10.16"
+  checksum="7601f844de6d906235140d0f76cca58be7ac716f3e2c29c35845aa24b24f73b9"
+  if [[ $(uname -m) == 'arm64' ]]
```

Comment:
Are we planning on having 2 mac builds - both arm and x64? I would assume not, in which case, we don't really need this check.

Reply:
I think it's not really decided yet. However, maybe some users may use this script to setup the dependencies, some may still have x86.

Reply:
I really doubt anyone uses this to set up dependencies locally - but it does make sense to at least discuss keeping both builds for a while. E.g., I have an x64 Macbook, although I'm sure I'm in the minority :)

Reply:
We said we'd look into how much hassle fat builds would be first...

Reply:
Not sure how osx deals with dynamic libraries there though.

Reply:
Fat builds?

Reply:
universal binaries that work across platforms natively. That should in theory just be a flag to xcode.

Reply:
Although it remains to be seen how much hassle it'd be in practice.

Reply:
Ah, didn't know they were referred to as 'fat' builds.

Reply:
Yeah, these days I guess universal binaries is more common

Reply:
ok, this PR implements now the creation of universal binaries... however, what about testing? right now this only tests the apple silicon part of the universal binaries..